# DATA ANALYSIS AND FORECASTING OF SINALOA'S FISH PRODUCTION

## 1. ABOUT THE PROJECT

**Author**: Fernando A. Martinez Urrutia

As a Data Analysts | Scientist we use data as our source of power. If knowledge is power and data is knowledge then by logic data is power!…But **every power comes with great responsibility**. Lets imagine for a moment that you're the secretariat of natural resources of your country…what would be the first thing you would like to know about the different sectors that you manage? What decisions would you make based on the data presented to you? What programs would you invest or stop investing in?…THAT is the power that data can have on our life.


**Objective**: 
In this project I'll analyze the mexican fishing industry over the period of 2017-2020 in order to detect major trends and potential growth opportunities. After an intial exploratory data analysis, an univariate Machine Learning Model will be selected to best predict the monetary production of the most productive state.

**Questions to answer:**
- Which state is the one with the biggest fish production
- What sector is more profitable: fishing or aquaculture
- Most profitable family of fish in Mexico

**About the data:** This data set is composed by 4 csv tables from the fishing production from the years 2017 to 2020 reported by CONAPESCA (National Comission of Fishing and Aquaculture). Giving a total dataframe of 312,700 entries, each with 12 attributes.
*"Produccion Pesquera, Datos y Recursos", published by CONAPESCA. Obtained from https://datos.gob.mx/busca/dataset/produccion-pesquera. the 2021-01-20*

**Tools used:** Exploratory Data Analysis, Data Cleaning, Data Visualization.

regex, pandas, numpy, v

##  2. DATA PREPROCESS

### 2.1. Data Extraction

For our analysis we will use data from the years 2017 to 2020. Please download the .csv files from the provided references and load it to your jupiter notebook or to Visual Studio Code; upload the files to a folder named "data" in your workspace

In [1]:
import pandas as pd
import numpy as np

we now turn the CSV files to dataframes using the pandas library and concatenated them into a single dataframe called "df"

In [2]:
df2017=pd.read_csv('../input/mexican-fishing-industry-2017-2020/Produccion_Pesquera_2017.csv',encoding='latin-1')
df2018=pd.read_csv('../input/mexican-fishing-industry-2017-2020/Produccion_Pesquera_2018.csv',encoding='latin-1')
df2019=pd.read_csv('../input/mexican-fishing-industry-2017-2020/Produccion_Pesquera_2019.csv',encoding='latin-1')
df2020=pd.read_csv('../input/mexican-fishing-industry-2017-2020/Produccion_Pesquera_2020.csv',encoding='latin-1')
df=pd.concat([df2017, df2018,df2019,df2020],axis=0)
display(df.head(20))


/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3552: DtypeWarning: Columns (1,2,3,4,5,6,8,9) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)
/opt/conda/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3552: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


EJERCICIO FISCAL   ENTIDAD FEDERATIVA         NOMBRE OFICINA MES DE CORTE  \
0             2017.0      BAJA CALIFORNIA             SAN FELIPE        JULIO   
1             2017.0      BAJA CALIFORNIA             SAN FELIPE        JULIO   
2             2017.0  BAJA CALIFORNIA SUR         BAHIA ASUNCION       AGOSTO   
3             2017.0  BAJA CALIFORNIA SUR          SANTA ROSALIA    NOVIEMBRE   
4             2017.0               COLIMA                TECOMAN      FEBRERO   
5             2017.0             CAMPECHE                 ATASTA    NOVIEMBRE   
6             2017.0             CAMPECHE               CAMPECHE        ENERO   
7             2017.0             CAMPECHE               CAMPECHE        ABRIL   
8             2017.0             CAMPECHE               CAMPECHE         MAYO   
9             2017.0             CAMPECHE              CHAMPOTON         MAYO   
10            2017.0             CAMPECHE              CHAMPOTON        JUNIO   
11            2017.0             CAMPECHE             ISLA ARENA   SEPTIEMBRE   
12            2017.0             CAMPECHE             ISLA ARENA    NOVIEMBRE   
13            2017.0             CAMPECHE               SABANCUY      OCTUBRE   
14            2017.0             CAMPECHE             SEYBAPLAYA        ENERO   
15            2017.0             GUERRERO  MARQUELIA CRUZ GRANDE    NOVIEMBRE   
16            2017.0             GUERRERO  MARQUELIA CRUZ GRANDE    NOVIEMBRE   
17            2017.0              JALISCO        PUERTO VALLARTA      FEBRERO   
18            2017.0              JALISCO        PUERTO VALLARTA        ABRIL   
19            2017.0              JALISCO        PUERTO VALLARTA         MAYO   

     ORIGEN NOMBRE COMUN PESO DESEMBARCADO KILOGRAMOS PESO VIVO KILOGRAMOS  \
0   CAPTURA      ABADEJO                           94                 94.0   
1   CAPTURA      ABADEJO                         1400               1540.0   
2   CAPTURA      ABADEJO                         4397               4397.0   
3   CAPTURA      ABADEJO                           17                 17.0   
4   CAPTURA      ABADEJO                         4587               4587.0   
5   CAPTURA      ABADEJO                         4406               4406.0   
6   CAPTURA      ABADEJO                           18                 18.0   
7   CAPTURA      ABADEJO                          103                103.0   
8   CAPTURA      ABADEJO                          189                189.0   
9   CAPTURA      ABADEJO                           52                 52.0   
10  CAPTURA      ABADEJO                           37                 37.0   
11  CAPTURA      ABADEJO                         2280               2280.0   
12  CAPTURA      ABADEJO                           13                 13.0   
13  CAPTURA      ABADEJO                           77                 77.0   
14  CAPTURA      ABADEJO                        40016              40016.0   
15  CAPTURA      ABADEJO                          221                221.0   
16  CAPTURA      ABADEJO                         1100               1210.0   
17  CAPTURA      ABADEJO                          155                155.0   
18  CAPTURA      ABADEJO                          240                240.0   
19  CAPTURA      ABADEJO                          430                430.0   

    VALOR PESOS NOMBRE PRINCIPAL  CLAVE DE ENTIDAD  CLAVE DE OFICINA  
0     3119.6928             MERO               2.0             207.0  
1         46424             MERO               2.0             207.0  
2     159567.13             MERO               3.0             313.0  
3        616.93             MERO               3.0             312.0  
4     166462.23             MERO               6.0             602.0  
5   159899.1835             MERO               4.0             410.0  
6      611.6664             MERO               4.0             401.0  
7     3525.9621             MERO               4.0             401.0  
8     6578.2216         

### 2.2 Data Cleaning

#### 2.2.1 DataFrame Insight

The first step we want to take is to known our data.
what type of data we have?, how much data we have? and to obtain it's main metrics. 

In [3]:
display (df.shape)
display(df.info(show_counts=True))
display(df.describe(include='all'))

(312752, 12)

<class 'pandas.core.frame.DataFrame'>
Int64Index: 312752 entries, 0 to 63235
Data columns (total 12 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   EJERCICIO FISCAL              251440 non-null  float64
 1   ENTIDAD FEDERATIVA            251440 non-null  object 
 2   NOMBRE OFICINA                251440 non-null  object 
 3   MES DE CORTE                  251440 non-null  object 
 4   ORIGEN                        251440 non-null  object 
 5   NOMBRE COMUN                  251440 non-null  object 
 6   PESO DESEMBARCADO KILOGRAMOS  251440 non-null  object 
 7   PESO VIVO KILOGRAMOS          251440 non-null  object 
 8   VALOR PESOS                   251440 non-null  object 
 9   NOMBRE PRINCIPAL              251440 non-null  object 
 10  CLAVE DE ENTIDAD              251440 non-null  float64
 11  CLAVE DE OFICINA              251440 non-null  float64
dtypes: float64(3), object(9)
memory usage: 31.0+ 

None

EJERCICIO FISCAL   ENTIDAD FEDERATIVA NOMBRE OFICINA MES DE CORTE  \
count      251440.000000               251440         251440       251440   
unique               NaN                   31            157           12   
top                  NaN  BAJA CALIFORNIA SUR       ENSENADA        MARZO   
freq                 NaN                31614           5571        22339   
mean         2018.489449                  NaN            NaN          NaN   
std             1.124732                  NaN            NaN          NaN   
min          2017.000000                  NaN            NaN          NaN   
25%          2017.000000                  NaN            NaN          NaN   
50%          2018.000000                  NaN            NaN          NaN   
75%          2020.000000                  NaN            NaN          NaN   
max          2020.000000                  NaN            NaN          NaN   

         ORIGEN NOMBRE COMUN PESO DESEMBARCADO KILOGRAMOS  \
count    251440       251440                       251440   
unique        2          675                        67919   
top     CAPTURA   TILAPIA SE                          100   
freq     230078        10935                         1438   
mean        NaN          NaN                          NaN   
std         NaN          NaN                          NaN   
min         NaN          NaN                          NaN   
25%         NaN          NaN                          NaN   
50%         NaN          NaN                          NaN   
75%         NaN          NaN                          NaN   
max         NaN          NaN                          NaN   

       PESO VIVO KILOGRAMOS VALOR PESOS NOMBRE PRINCIPAL  CLAVE DE ENTIDAD  \
count                251440      251440           251440     251440.000000   
unique                77851      170970               57               NaN   
top                       0       6,000            OTRAS               NaN   
freq                   1250         247            48999               NaN   
mean                    NaN         NaN              NaN         17.789373   
std                     NaN         NaN              NaN         10.575871   
min                     NaN         NaN              NaN          1.000000   
25%                     NaN         NaN              NaN          4.000000   
50%                     NaN         NaN              NaN         20.000000   
75%                     NaN         NaN              NaN         27.000000   
max                     NaN         NaN              NaN         32.000000   

        CLAVE DE OFICINA  
count      251440.000000  
unique               NaN  
top                  NaN  
freq                 NaN  
mean         1784.014067  
std          1058.024604  
min           101.000000  
25%           410.000000  
50%          2004.000000  
75%          2705.000000  
max          3201.000000

- note how we have 312,752 rows but only 251,440 from those are not null.
- Summary stadistics were calculated for 'EJERCICIO FISCAL','CLAVE DE OFICINA' and 'CLAVE DE ENTIDAD' although they should be categorical values.
- Note how we have 2 type of columns objects and float values. 
- The stadistical descriptors were not calculated for the numerical columns, meaning they must contain mixed data or strings; this will be solved in the following cells. 

#### 2.2.2 Renaming columns and dropping columns that will not be used

In [4]:
#rename columns and data frame and deletion of the 2 last columns which will not be used. 
df = df.rename(columns={'ENTIDAD FEDERATIVA':'STATE','EJERCICIO FISCAL':'YEAR','NOMBRE OFICINA':'OFFICE',"NOMBRE PRINCIPAL":'FAMILY',"NOMBRE COMUN":'SPECIE',"MES DE CORTE":"MONTH","VALOR PESOS":"PROFIT","PESO VIVO KILOGRAMOS":"LIVE_CAPTURED_KG","PESO DESEMBARCADO KILOGRAMOS":"UNEMBARKED_KG","ORIGEN":"ORIGIN"}).drop(columns=['CLAVE DE ENTIDAD','CLAVE DE OFICINA'])
df.head(10)

YEAR                STATE          OFFICE      MONTH   ORIGIN   SPECIE  \
0  2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO  CAPTURA  ABADEJO   
1  2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO  CAPTURA  ABADEJO   
2  2017.0  BAJA CALIFORNIA SUR  BAHIA ASUNCION     AGOSTO  CAPTURA  ABADEJO   
3  2017.0  BAJA CALIFORNIA SUR   SANTA ROSALIA  NOVIEMBRE  CAPTURA  ABADEJO   
4  2017.0               COLIMA         TECOMAN    FEBRERO  CAPTURA  ABADEJO   
5  2017.0             CAMPECHE          ATASTA  NOVIEMBRE  CAPTURA  ABADEJO   
6  2017.0             CAMPECHE        CAMPECHE      ENERO  CAPTURA  ABADEJO   
7  2017.0             CAMPECHE        CAMPECHE      ABRIL  CAPTURA  ABADEJO   
8  2017.0             CAMPECHE        CAMPECHE       MAYO  CAPTURA  ABADEJO   
9  2017.0             CAMPECHE       CHAMPOTON       MAYO  CAPTURA  ABADEJO   

  UNEMBARKED_KG LIVE_CAPTURED_KG       PROFIT FAMILY  
0            94             94.0    3119.6928   MERO  
1          1400           1540.0        46424   MERO  
2          4397           4397.0    159567.13   MERO  
3            17             17.0       616.93   MERO  
4          4587           4587.0    166462.23   MERO  
5          4406           4406.0  159899.1835   MERO  
6            18             18.0     611.6664   MERO  
7           103            103.0    3525.9621   MERO  
8           189            189.0    6578.2216   MERO  
9            52             52.0     1767.647   MERO

#### 2.2.3 Treating missing values and Mixed Dtypes

Given that somes numerical columns are expressed with commas, they will be read as strings. That is the reason behind why the previous stadistical descriptors did not work in the columns, to solve this problem we can remove all commas from numbers using **regex**, transform them to float values and drop the entries with NaN values.

In [5]:
#drop rows in which all data is missing and reset the index
df=df.dropna(how='all').reset_index(drop=True)
display(df)
df = df.replace(',','', regex=True)
display(df)
# transforming string or float values to numbers
cols = ['LIVE_CAPTURED_KG', 'UNEMBARKED_KG', 'PROFIT','YEAR']
df[cols] = df[cols].apply(pd.to_numeric, errors='coerce', axis=1)
#replace NaN with O values
df['YEAR'] = df['YEAR'].replace(np.nan, 0)
df['UNEMBARKED_KG'] = df["UNEMBARKED_KG"].replace(np.nan, 0)
df['LIVE_CAPTURED_KG']= df['LIVE_CAPTURED_KG'].replace(np.nan, 0)
df['PROFIT'] = df['PROFIT'].replace(np.nan, 0)

YEAR                STATE          OFFICE      MONTH       ORIGIN  \
0       2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO      CAPTURA   
1       2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO      CAPTURA   
2       2017.0  BAJA CALIFORNIA SUR  BAHIA ASUNCION     AGOSTO      CAPTURA   
3       2017.0  BAJA CALIFORNIA SUR   SANTA ROSALIA  NOVIEMBRE      CAPTURA   
4       2017.0               COLIMA         TECOMAN    FEBRERO      CAPTURA   
...        ...                  ...             ...        ...          ...   
251435  2020.0               PUEBLA          PUEBLA     AGOSTO  ACUACULTURA   
251436  2020.0               PUEBLA          PUEBLA    OCTUBRE  ACUACULTURA   
251437  2020.0               PUEBLA          PUEBLA  NOVIEMBRE  ACUACULTURA   
251438  2020.0               PUEBLA          PUEBLA  DICIEMBRE  ACUACULTURA   
251439  2020.0             VERACRUZ        VERACRUZ      MARZO  ACUACULTURA   

           SPECIE UNEMBARKED_KG LIVE_CAPTURED_KG     PROFIT  FAMILY  
0         ABADEJO            94             94.0  3119.6928    MERO  
1         ABADEJO          1400           1540.0      46424    MERO  
2         ABADEJO          4397           4397.0  159567.13    MERO  
3         ABADEJO            17             17.0     616.93    MERO  
4         ABADEJO          4587           4587.0  166462.23    MERO  
...           ...           ...              ...        ...     ...  
251435  TRUCHA SE         1,984            1,984    150,139  TRUCHA  
251436  TRUCHA SE         3,100            3,100    211,463  TRUCHA  
251437  TRUCHA SE           484              484     36,477  TRUCHA  
251438  TRUCHA SE        16,615           16,615  1,208,277  TRUCHA  
251439  TRUCHA SE           248              248     17,340  TRUCHA  

[251440 rows x 10 columns]

YEAR                STATE          OFFICE      MONTH       ORIGIN  \
0       2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO      CAPTURA   
1       2017.0      BAJA CALIFORNIA      SAN FELIPE      JULIO      CAPTURA   
2       2017.0  BAJA CALIFORNIA SUR  BAHIA ASUNCION     AGOSTO      CAPTURA   
3       2017.0  BAJA CALIFORNIA SUR   SANTA ROSALIA  NOVIEMBRE      CAPTURA   
4       2017.0               COLIMA         TECOMAN    FEBRERO      CAPTURA   
...        ...                  ...             ...        ...          ...   
251435  2020.0               PUEBLA          PUEBLA     AGOSTO  ACUACULTURA   
251436  2020.0               PUEBLA          PUEBLA    OCTUBRE  ACUACULTURA   
251437  2020.0               PUEBLA          PUEBLA  NOVIEMBRE  ACUACULTURA   
251438  2020.0               PUEBLA          PUEBLA  DICIEMBRE  ACUACULTURA   
251439  2020.0             VERACRUZ        VERACRUZ      MARZO  ACUACULTURA   

           SPECIE UNEMBARKED_KG LIVE_CAPTURED_KG     PROFIT  FAMILY  
0         ABADEJO            94             94.0  3119.6928    MERO  
1         ABADEJO          1400           1540.0      46424    MERO  
2         ABADEJO          4397           4397.0  159567.13    MERO  
3         ABADEJO            17             17.0     616.93    MERO  
4         ABADEJO          4587           4587.0  166462.23    MERO  
...           ...           ...              ...        ...     ...  
251435  TRUCHA SE          1984             1984     150139  TRUCHA  
251436  TRUCHA SE          3100             3100     211463  TRUCHA  
251437  TRUCHA SE           484              484      36477  TRUCHA  
251438  TRUCHA SE         16615            16615    1208277  TRUCHA  
251439  TRUCHA SE           248              248      17340  TRUCHA  

[251440 rows x 10 columns]

In [6]:
print("NaN in YEAR",df['YEAR'].isnull().sum())
print("NaN in UNEMBARKED_KG",df['UNEMBARKED_KG'].isnull().sum())
print("NaN in LIVE_CAPTURED_KG",df['LIVE_CAPTURED_KG'].isnull().sum())
print("NaN in PROFIT",df['PROFIT'].isnull().sum())
print("NaN in YEAR",df['YEAR'].isnull().values.any())
print("NaN in UNEMBARKED_KG",df['UNEMBARKED_KG'].isnull().values.any())
print("NaN in LIVE_CAPTURED_KG",df['LIVE_CAPTURED_KG'].isnull().values.any())
print("NaN in PROFIT",df['PROFIT'].isnull().values.any())


NaN in YEAR 0
NaN in UNEMBARKED_KG 0
NaN in LIVE_CAPTURED_KG 0
NaN in PROFIT 0
NaN in YEAR False
NaN in UNEMBARKED_KG False
NaN in LIVE_CAPTURED_KG False
NaN in PROFIT False


**classify of columns between categories and numbers**

In [7]:
#transform numerical colums to int64
columns=['LIVE_CAPTURED_KG', 'UNEMBARKED_KG', 'PROFIT']
df[columns]=df[columns].astype(np.int64)
#transform objects to categories
for col in ['STATE', 'MONTH', 'ORIGIN','SPECIE','FAMILY','OFFICE']:df[col] = df[col].astype("category")
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 251440 entries, 0 to 251439
Data columns (total 10 columns):
 #   Column            Non-Null Count   Dtype   
---  ------            --------------   -----   
 0   YEAR              251440 non-null  float64 
 1   STATE             251440 non-null  category
 2   OFFICE            251440 non-null  category
 3   MONTH             251440 non-null  category
 4   ORIGIN            251440 non-null  category
 5   SPECIE            251440 non-null  category
 6   UNEMBARKED_KG     251440 non-null  int64   
 7   LIVE_CAPTURED_KG  251440 non-null  int64   
 8   PROFIT            251440 non-null  int64   
 9   FAMILY            251440 non-null  category
dtypes: category(6), float64(1), int64(3)
memory usage: 9.6 MB


None

#### 2.2.4 turning dataframe into time series

In [9]:
#turn  strings of month to format 01-10 to later concatenate with year column
df["MONTH"].replace({'ENERO':"01-24","FEBRERO":"02-24","MARZO":"03-24","ABRIL":'04-24',"MAYO":"05-24","JUNIO":"06-24","JULIO":"07-24","AGOSTO":"08-24","SEPTIEMBRE":"09-24","OCTUBRE":"10-24","NOVIEMBRE":"11-24","DICIEMBRE":"12-24"},inplace=True)
#convert float values of year to int and latter to interger, as we cannot convert floats "2017.0" to string directly
df["YEAR"]=df["YEAR"].astype(np.int64)
#convert year column to string to be able to concatenate with the month and day
df["YEAR"]=df["YEAR"].astype(str)
#concatenate month and date with year column
df["DATE"]=df[["YEAR","MONTH"]].apply(lambda x: "-".join(x), axis =1)
# turn to timeseries format
df["DATE"]= pd.to_datetime(df["DATE"],format='%Y-%m-%d', errors='coerce')
# even though we have the date in timeseries format, for the exploratory analysis 
# we'll be keeping the year and month in separate columnsn which will faciliate the analysis
# as we will set the date as our index 
df["YEAR"]=df["DATE"].dt.year
df["MONTH"]=df["DATE"].dt.month
df=df.set_index("DATE",drop=False)
display(df)

YEAR                STATE          OFFICE  MONTH       ORIGIN  \
DATE                                                                        
2017-07-24  2017      BAJA CALIFORNIA      SAN FELIPE      7      CAPTURA   
2017-07-24  2017      BAJA CALIFORNIA      SAN FELIPE      7      CAPTURA   
2017-08-24  2017  BAJA CALIFORNIA SUR  BAHIA ASUNCION      8      CAPTURA   
2017-11-24  2017  BAJA CALIFORNIA SUR   SANTA ROSALIA     11      CAPTURA   
2017-02-24  2017               COLIMA         TECOMAN      2      CAPTURA   
...          ...                  ...             ...    ...          ...   
2020-08-24  2020               PUEBLA          PUEBLA      8  ACUACULTURA   
2020-10-24  2020               PUEBLA          PUEBLA     10  ACUACULTURA   
2020-11-24  2020               PUEBLA          PUEBLA     11  ACUACULTURA   
2020-12-24  2020               PUEBLA          PUEBLA     12  ACUACULTURA   
2020-03-24  2020             VERACRUZ        VERACRUZ      3  ACUACULTURA   

               SPECIE  UNEMBARKED_KG  LIVE_CAPTURED_KG   PROFIT  FAMILY  \
DATE                                                                      
2017-07-24    ABADEJO             94                94     3119    MERO   
2017-07-24    ABADEJO           1400              1540    46424    MERO   
2017-08-24    ABADEJO           4397              4397   159567    MERO   
2017-11-24    ABADEJO             17                17      616    MERO   
2017-02-24    ABADEJO           4587              4587   166462    MERO   
...               ...            ...               ...      ...     ...   
2020-08-24  TRUCHA SE           1984              1984   150139  TRUCHA   
2020-10-24  TRUCHA SE           3100              3100   211463  TRUCHA   
2020-11-24  TRUCHA SE            484               484    36477  TRUCHA   
2020-12-24  TRUCHA SE          16615             16615  1208277  TRUCHA   
2020-03-24  TRUCHA SE            248               248    17340  TRUCHA   

                 DATE  
DATE                   
2017-07-24 2017-07-24  
2017-07-24 2017-07-24  
2017-08-24 2017-08-24  
2017-11-24 2017-11-24  
2017-02-24 2017-02-24  
...               ...  
2020-08-24 2020-08-24  
2020-10-24 2020-10-24  
2020-11-24 2020-11-24  
2020-12-24 2020-12-24  
2020-03-24 2020-03-24  

[251440 rows x 11 columns]

In [11]:
df_fishing_2017_2020=df
df_fishing_2017_2020.describe(include="all")

/opt/conda/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  


YEAR                STATE    OFFICE          MONTH   ORIGIN  \
count   251440.000000               251440    251440  251440.000000   251440   
unique            NaN                   31       157            NaN        2   
top               NaN  BAJA CALIFORNIA SUR  ENSENADA            NaN  CAPTURA   
freq              NaN                31614      5571            NaN   230078   
first             NaN                  NaN       NaN            NaN      NaN   
last              NaN                  NaN       NaN            NaN      NaN   
mean      2018.489449                  NaN       NaN       6.469830      NaN   
std          1.124732                  NaN       NaN       3.496539      NaN   
min       2017.000000                  NaN       NaN       1.000000      NaN   
25%       2017.000000                  NaN       NaN       3.000000      NaN   
50%       2018.000000                  NaN       NaN       7.000000      NaN   
75%       2020.000000                  NaN       NaN      10.000000      NaN   
max       2020.000000                  NaN       NaN      12.000000      NaN   

            SPECIE  UNEMBARKED_KG  LIVE_CAPTURED_KG        PROFIT  FAMILY  \
count       251440   2.514400e+05      2.514400e+05  2.514400e+05  251440   
unique         675            NaN               NaN           NaN      57   
top     TILAPIA SE            NaN               NaN           NaN   OTRAS   
freq         10935            NaN               NaN           NaN   48999   
first          NaN            NaN               NaN           NaN     NaN   
last           NaN            NaN               NaN           NaN     NaN   
mean           NaN   2.989888e+04      3.241845e+04  6.668708e+05     NaN   
std            NaN   4.191346e+05      4.635418e+05  9.506821e+06     NaN   
min            NaN   0.000000e+00      0.000000e+00  0.000000e+00     NaN   
25%            NaN   2.790000e+02      2.890000e+02  6.454000e+03     NaN   
50%            NaN   1.318000e+03      1.386000e+03  3.198550e+04     NaN   
75%            NaN   6.100000e+03      6.471000e+03  1.537000e+05     NaN   
max            NaN   5.789263e+07      5.789263e+07  1.300189e+09     NaN   

                       DATE  
count                251440  
unique                   48  
top     2017-03-24 00:00:00  
freq                   5823  
first   2017-01-24 00:00:00  
last    2020-12-24 00:00:00  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

As we can observe now only the numerical columns are the ones with statistical descriptors

### 2.3 Processed Dataframe file Saving

In [12]:
df_fishing_2017_2020.to_csv("df_fishing_2017_2020.csv", index=False)

## 3. Exploratory Data Analysis

In [13]:
import pandas as pd
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%pip install matplotlib

Note: you may need to restart the kernel to use updated packages.


In [14]:
display(df_fishing_2017_2020.describe(include="all"))
display(df_fishing_2017_2020)

YEAR                STATE          OFFICE  MONTH       ORIGIN  \
DATE                                                                        
2017-07-24  2017      BAJA CALIFORNIA      SAN FELIPE      7      CAPTURA   
2017-07-24  2017      BAJA CALIFORNIA      SAN FELIPE      7      CAPTURA   
2017-08-24  2017  BAJA CALIFORNIA SUR  BAHIA ASUNCION      8      CAPTURA   
2017-11-24  2017  BAJA CALIFORNIA SUR   SANTA ROSALIA     11      CAPTURA   
2017-02-24  2017               COLIMA         TECOMAN      2      CAPTURA   
...          ...                  ...             ...    ...          ...   
2020-08-24  2020               PUEBLA          PUEBLA      8  ACUACULTURA   
2020-10-24  2020               PUEBLA          PUEBLA     10  ACUACULTURA   
2020-11-24  2020               PUEBLA          PUEBLA     11  ACUACULTURA   
2020-12-24  2020               PUEBLA          PUEBLA     12  ACUACULTURA   
2020-03-24  2020             VERACRUZ        VERACRUZ      3  ACUACULTURA   

               SPECIE  UNEMBARKED_KG  LIVE_CAPTURED_KG   PROFIT  FAMILY  \
DATE                                                                      
2017-07-24    ABADEJO             94                94     3119    MERO   
2017-07-24    ABADEJO           1400              1540    46424    MERO   
2017-08-24    ABADEJO           4397              4397   159567    MERO   
2017-11-24    ABADEJO             17                17      616    MERO   
2017-02-24    ABADEJO           4587              4587   166462    MERO   
...               ...            ...               ...      ...     ...   
2020-08-24  TRUCHA SE           1984              1984   150139  TRUCHA   
2020-10-24  TRUCHA SE           3100              3100   211463  TRUCHA   
2020-11-24  TRUCHA SE            484               484    36477  TRUCHA   
2020-12-24  TRUCHA SE          16615             16615  1208277  TRUCHA   
2020-03-24  TRUCHA SE            248               248    17340  TRUCHA   

                 DATE  
DATE                   
2017-07-24 2017-07-24  
2017-07-24 2017-07-24  
2017-08-24 2017-08-24  
2017-11-24 2017-11-24  
2017-02-24 2017-02-24  
...               ...  
2020-08-24 2020-08-24  
2020-10-24 2020-10-24  
2020-11-24 2020-11-24  
2020-12-24 2020-12-24  
2020-03-24 2020-03-24  

[251440 rows x 11 columns]

In [18]:
profit=df_fishing_2017_2020.groupby(["YEAR","MONTH"]).sum()
s= profit.reset_index().pivot(index='DATE',columns="MONTH", values='PROFIT')
sns.heatmap(s,cbar_kws={"label":'value in pesos'})
plt.title("Mexican fishing production 2017-2020 (pesos)")
plt.xlabel("month")
plt.ylabel("year")
plt.show()
display(s)

KeyError: 'DATE'